source/client_channel_handler.c

```diff
@@ -50,9 +50,16 @@ static int s_packet_handler_connack(
         return AWS_OP_ERR;
     }
 
-    connection->state = AWS_MQTT_CLIENT_STATE_CONNECTED;
+    if (connack.connect_return_code == AWS_MQTT_CONNECT_ACCEPTED) {
+        connection->state = AWS_MQTT_CLIENT_STATE_CONNECTED;
+    }
 
-    MQTT_CLIENT_CALL_CALLBACK(connection, on_connack, connack.connect_return_code, connack.session_present);
+    if (connection->state == AWS_MQTT_CLIENT_STATE_RECONNECTING) {
+        MQTT_CLIENT_CALL_CALLBACK_ARGS(connection, on_resumed, connack.connect_return_code, connack.session_present);
+    } else {
+        MQTT_CLIENT_CALL_CALLBACK_ARGS(
+            connection, on_connection_complete, AWS_OP_SUCCESS, connack.connect_return_code, connack.session_present);
```

Comment:
If someone's called disconnect() from the RECONNECTING state, that will put us into DISCONNECTING state and the on_connection_complete() callback will fire a 2nd time.